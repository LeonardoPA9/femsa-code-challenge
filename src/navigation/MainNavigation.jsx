import React from "react";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { NavigationContainer } from "@react-navigation/native";
import MainScreen from "../screens/MainScreen";
import { RouteNames } from "../utils/constants/routes";
import ProductDetails from "../screens/ProductDetails";
import { GlobalStyles } from "../utils/constants/colors";

const { Navigator, Screen } = createNativeStackNavigator();

const MainNavigation = () => {
  return (
    <NavigationContainer>
      <Navigator
        screenOptions={{
          contentStyle: { backgroundColor: GlobalStyles.colors.background },
        }}
        initialRouteName={RouteNames.HOME}
      >
        <Screen
          options={{ headerShown: false }}
          name={RouteNames.HOME}
          component={MainScreen}
        />
        <Screen
          options={{
            title: "nombre del producto",
            headerBackVisible: false,
            headerLargeTitle: true,
            headerStyle: {
              backgroundColor: GlobalStyles.colors.accent,
            },
            headerLargeTitleStyle: { fontSize: 32 },
          }}
          name={RouteNames.PRODUCT_DETAILS}
          component={ProductDetails}
        />
      </Navigator>
    </NavigationContainer>
  );
};

export default MainNavigation;
