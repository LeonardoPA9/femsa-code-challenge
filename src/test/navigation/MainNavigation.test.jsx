import React from "react";
import { render } from "@testing-library/react-native";
import MainNavigation from "../../navigation/MainNavigation";

describe("MainNavigation", () => {
  it("matches snapshot", () => {
    const { toJSON } = render(<MainNavigation />);
    expect(toJSON()).toMatchSnapshot();
  });
});
