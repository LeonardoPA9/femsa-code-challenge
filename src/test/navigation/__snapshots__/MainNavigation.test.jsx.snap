// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavigation matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              {
                "backgroundColor": "#F8F8F8",
              },
            ]
          }
        >
          <RCTSafeAreaView
            style={
              {
                "flex": 1,
                "paddingTop": 0,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                  "paddingHorizontal": 24,
                }
              }
            >
              <View
                style={
                  {
                    "marginTop": 30,
                  }
                }
              >
                <Text
                  style={
                    {
                      "fontFamily": "avenir-bold",
                      "fontSize": 30,
                      "lineHeight": 39,
                    }
                  }
                >
                  Bienvenido de vuelta!
                </Text>
                <Text
                  style={
                    {
                      "fontFamily": "avenir",
                      "fontSize": 21,
                      "lineHeight": 27,
                    }
                  }
                >
                  Leonardo Pesina
                </Text>
              </View>
              <View
                style={
                  {
                    "marginTop": 25,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontFamily": "avenir-bold",
                        "fontSize": 30,
                        "lineHeight": 39,
                      },
                      {
                        "color": "#9B9898",
                        "fontSize": 18,
                        "lineHeight": 25,
                      },
                    ]
                  }
                >
                  TUS PUNTOS
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "paddingTop": 20,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "backgroundColor": "#334FFA",
                        "borderRadius": 20,
                        "elevation": 20,
                        "height": 143,
                        "marginHorizontal": 53,
                        "paddingHorizontal": 19,
                        "paddingVertical": 21,
                        "shadowColor": "gba(0, 0, 0, 0.5)",
                        "shadowOffset": {
                          "height": 4,
                          "width": 0,
                        },
                        "shadowOpacity": 0.5,
                        "shadowRadius": 4,
                        "width": 300,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontFamily": "avenir-bold",
                            "fontSize": 30,
                            "lineHeight": 39,
                          },
                          {
                            "color": "#ffffff",
                            "fontSize": 16,
                            "lineHeight": 22,
                          },
                        ]
                      }
                    >
                      Diciembre
                    </Text>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flex": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "fontFamily": "avenir-bold",
                              "fontSize": 30,
                              "lineHeight": 39,
                            },
                            {
                              "color": "#ffffff",
                              "fontSize": 32,
                              "lineHeight": 44,
                            },
                          ]
                        }
                      >
                        0.00 pts
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  {
                    "marginTop": 25,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontFamily": "avenir-bold",
                        "fontSize": 30,
                        "lineHeight": 39,
                      },
                      {
                        "color": "#9B9898",
                        "fontSize": 18,
                        "lineHeight": 25,
                      },
                    ]
                  }
                >
                  TUS MOVIMIENTOS
                </Text>
                <View
                  style={
                    {
                      "backgroundColor": "#ffffff",
                      "borderRadius": 15,
                      "height": 275,
                      "marginTop": 20,
                      "padding": 20,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTScrollView
                      applyWindowCorrection={[Function]}
                      canChangeSize={true}
                      contentContainerStyle={
                        {
                          "backgroundColor": undefined,
                          "minHeight": 1,
                          "minWidth": 1,
                          "paddingLeft": 0,
                          "paddingRight": 0,
                        }
                      }
                      contentHeight={0}
                      contentWidth={0}
                      data={[]}
                      dataProvider={
                        DataProvider {
                          "_data": [],
                          "_firstIndexToProcess": 0,
                          "_hasStableIds": false,
                          "_requiresDataChangeHandling": false,
                          "_size": 0,
                          "getStableId": [Function],
                          "rowHasChanged": [Function],
                        }
                      }
                      disableRecycling={false}
                      estimatedItemSize={55}
                      extendedState={{}}
                      externalScrollView={[Function]}
                      finalRenderAheadOffset={250}
                      forceNonDeterministicRendering={true}
                      horizontal={false}
                      initialOffset={0}
                      initialRenderIndex={0}
                      isHorizontal={false}
                      layoutProvider={
                        GridLayoutProviderWithProps {
                          "_acceptableRelayoutDelta": 1,
                          "_getHeightOrWidth": [Function],
                          "_getLayoutTypeForIndex": [Function],
                          "_getSpan": [Function],
                          "_hasExpired": false,
                          "_maxSpan": 1,
                          "_setLayoutForType": [Function],
                          "_tempDim": {
                            "height": 0,
                            "width": 0,
                          },
                          "averageWindow": AverageWindow {
                            "currentAverage": 55,
                            "currentCount": 1,
                            "inputValues": [
                              55,
                            ],
                            "nextIndex": 1,
                          },
                          "defaultEstimatedItemSize": 100,
                          "layoutObject": {
                            "size": undefined,
                            "span": undefined,
                          },
                          "props": {
                            "data": [],
                            "estimatedItemSize": 55,
                            "numColumns": 1,
                            "renderItem": [Function],
                          },
                          "renderWindowInsets": {
                            "height": 0,
                            "width": 0,
                          },
                          "shouldRefreshWithAnchoring": true,
                        }
                      }
                      maxRenderAhead={750}
                      numColumns={1}
                      onEndReached={[Function]}
                      onEndReachedThreshold={0}
                      onEndReachedThresholdRelative={0}
                      onItemLayout={[Function]}
                      onLayout={[Function]}
                      onScroll={[Function]}
                      onScrollBeginDrag={[Function]}
                      onSizeChanged={[Function]}
                      onVisibleIndicesChanged={[Function]}
                      removeClippedSubviews={false}
                      renderAheadOffset={0}
                      renderAheadStep={250}
                      renderContentContainer={[Function]}
                      renderItem={[Function]}
                      renderItemContainer={[Function]}
                      rowRenderer={[Function]}
                      scrollEventThrottle={16}
                      scrollThrottle={16}
                      scrollViewProps={
                        {
                          "contentContainerStyle": {
                            "backgroundColor": undefined,
                            "minHeight": 1,
                            "minWidth": 1,
                            "paddingLeft": 0,
                            "paddingRight": 0,
                          },
                          "onLayout": [Function],
                          "onScrollBeginDrag": [Function],
                          "refreshControl": undefined,
                          "style": {
                            "minHeight": 1,
                            "minWidth": 1,
                          },
                        }
                      }
                      style={
                        {
                          "minHeight": 1,
                          "minWidth": 1,
                        }
                      }
                      suppressBoundedSizeException={true}
                      windowCorrectionConfig={
                        {
                          "applyToInitialOffset": true,
                          "applyToItemScroll": true,
                          "value": {
                            "endCorrection": 0,
                            "startCorrection": 0,
                            "windowShift": -0,
                          },
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            {
                              "flexDirection": "column",
                            }
                          }
                        >
                          <View
                            style={
                              {
                                "paddingLeft": 0,
                                "paddingTop": 0,
                              }
                            }
                          />
                          <View
                            style={
                              [
                                undefined,
                                undefined,
                              ]
                            }
                          />
                          <AutoLayoutView
                            enableInstrumentation={false}
                            horizontal={false}
                            onBlankAreaEvent={[Function]}
                            onLayout={[Function]}
                            renderAheadOffset={0}
                            scrollOffset={0}
                            style={
                              {
                                "height": 0,
                                "width": 0,
                              }
                            }
                            windowSize={0}
                          />
                          <CellContainer
                            index={-1}
                            style={
                              [
                                undefined,
                                undefined,
                              ]
                            }
                          />
                          <View
                            style={
                              {
                                "paddingBottom": 0,
                                "paddingRight": 0,
                              }
                            }
                          />
                        </View>
                      </View>
                    </RCTScrollView>
                  </View>
                </View>
              </View>
              <View
                style={
                  [
                    {
                      "marginTop": 25,
                    },
                    {
                      "columnGap": 10,
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <View
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={false}
                  >
                    <View
                      style={
                        {
                          "backgroundColor": "#334FFA",
                          "borderRadius": 10,
                          "paddingVertical": 15,
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "fontFamily": "avenir-bold",
                            },
                            {
                              "color": "#ffffff",
                              "textAlign": "center",
                              "textTransform": "capitalize",
                            },
                          ]
                        }
                      >
                        ganados
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={false}
                  >
                    <View
                      style={
                        {
                          "backgroundColor": "#334FFA",
                          "borderRadius": 10,
                          "paddingVertical": 15,
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "fontFamily": "avenir-bold",
                            },
                            {
                              "color": "#ffffff",
                              "textAlign": "center",
                              "textTransform": "capitalize",
                            },
                          ]
                        }
                      >
                        canjeados
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
